### DISCUSSION TITLE
How to read/write realm database created in react native via native/turbo/expo modules?

### DISCUSSION THREAD

**Discussion by BruceSuperProgramer** [realm/realm-js#5782] [Q&A]
Should I use realmjs in react native, realm-kotlin in android and realm-swift in ios?

Is there any documentation giving guidance on how to achieve this?

I found it really hard to do it I don't know where to start it.

Please help thank you.

**Comment by wmonecke** [comment]
@BruceSuperProgramer You are trying to have one DB that you can access from all sides (swift, java, and JS) right? Meaning one single source of truth?

**Reply by BruceSuperProgramer** [reply]
Yes, realmjs is write transaction is blocking UI. I'm thinking could we write data in another thread in native side(IOS/ Android).

**Reply by roni-castro-glooko** [reply]
Did you figure out how to do that?
> can access from all sides (swift, java, and JS)

**Reply by wesvh** [reply]
I have exactly the same impression, but I haven't achieved it, did you achieve it?

**Reply by BruceSuperProgramer** [reply]
Nope, haven't tried yet.

**Reply by AdarshJais** [reply]
Any solution yet?
